controllers/user_controller.go

```diff
@@ -109,3 +113,28 @@ func GetProfile() gin.HandlerFunc {
 		_ = result
 	}
 }
+
+func GetTrainer() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var input GetTrainerInput
+		if err := c.ShouldBindJSON(&input); err != nil {
+			c.JSON(http.StatusBadRequest, responses.GetTrainerResponses{
+				Status:  http.StatusBadRequest,
+				Message: "input missing"})
+			return
+		}
+		result, err := models.FindProfile(input.Username, "trainer")
+		if err != nil {
+			c.JSON(http.StatusBadRequest, responses.CurrentUserResponse{
+				Status:  http.StatusBadRequest,
+				Message: `trainer profile retrieval unsuccessful`,
+			})
+			return
+		}
+		c.JSON(http.StatusFound, responses.GetProfileResponses{
+			Status:  http.StatusFound,
+			Message: `Successfully retrieve trainer profile`,
+			User:    result,
+		})
```

Comment:
```suggestion
		c.JSON(http.StatusOK, responses.GetProfileResponses{
			Status:  http.StatusOK,
			Message: `Successfully retrieve trainer profile`,
			User:    result,
		})
```

Reply:
`StatusFound` is only used for redirect. You need to always return StatusOK for successful request.

Reply:
roger that

Reply:
Merging is done incorrectly please change to StatusOK